Modules/Module_Processing/Module_Population/xASL_stat_GetROIstatistics.m

```diff
@@ -754,13 +761,13 @@
 					end
 
                     %% CBF (now remove vascular artifacts)
-                    if x.S.bMasking(2)==1 % apply vascular mask
-                        CurrentMask = CurrentMask & VascularMask;
+                    if x.S.bMasking(2)==1 % apply vascular mask, but rename this to CurrentVascular mask, because we still need the original CurrentMask for 4D temporal calculations

```

Comment:
I like this, perhaps for clarity change this such that:

* rename CurrentMask -> CurrentMaskNotVascular
* rename CurrentVascularMask -> CurrentMaskVascular

to be sure. So we won't have CurrentMask anymore. Good idea?

As long as for CBF we always have CurrentMaskVascular (except when x.S.bMasking(2)==0 ) and for non-CBF (e.g. spatial CoV) we always have CurrentMaskNoVascular

That's probably why I did it by overwriting CurrentMask, that CurrentMask is always overwritten with new edits, rather than multiplying the number of CurrentMasks for each edits (which could lead to a lot of masks down the line)

Reply:
Done